### ISSUE TITLE
Container App job loses access to pull image from ACR after a period of time (still)

### ISSUE THREAD

**Issue by jason-berk-k1x** [microsoft/azure-container-apps#1204]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [ x ] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [ x ] regression (a behavior that used to work and stopped in a new release)

### Issue description
https://github.com/microsoft/azure-container-apps/issues/816#issuecomment-2181193648

### Steps to reproduce
no idea, that's part of the issue

**Expected behavior** [What you expected to happen.]

**Actual behavior** [What actually happened.]

**Screenshots**         

### Additional context
Ex. Did this issue occur in the CLI or the Portal?

**Comment by jason-berk-k1x** [comment]
not sure why the original issue closed a few days ago.  I literally ran into this exact issue today (6/20/24)

https://github.com/microsoft/azure-container-apps/issues/816#issuecomment-2181193648

**Comment by jason-berk-k1x** [comment]
I have since deleted my job and re-ran my terraform to create a new one.  this new job is using the `mcr.microsoft.com/k8se/quickstart-jobs:latest` image.  After the TF runs, I use the azure CLI to update the app based on a configuration yaml file.  That pipeline ran and in the portal everything _looks_ correct, but when I put a message on the queue that triggers the job, I see this in the system logs (and the execution fails)

![Screenshot 2024-06-20 at 3 18 56 PM](https://github.com/microsoft/azure-container-apps/assets/148882427/f17e8e7f-3a79-446b-9098-b77b78f054a1)

my job lives in the `dev` subscription and the ACR is in the `global` subscription but I've got many other jobs all built out the same way and none of them have this issue.

**Comment by jason-berk-k1x** [comment]
so, all my jobs were working just fine _before_ 6/20, at which point they all started failing.  Today, everything is working again....  Looking at the change history, seems Azure updated my environments KEDA version right before everything broke....and reverted it Sunday and today on Monday everything works:

![Screenshot 2024-06-24 at 10 10 58 AM](https://github.com/microsoft/azure-container-apps/assets/148882427/34f26aa0-ee5c-4124-a834-79c01e07139e)

not sure why the KEDA version change would cause the issues I'm seeing, but the timing seem very peculiar....

**Comment by jason-berk-k1x** [comment]
and I found this: https://github.com/microsoft/azure-container-apps/issues/1207#issuecomment-2183792297

**Comment by fritzfs** [comment]
@jason-berk-k1x did you resolve this? Today we encountered [this issue](https://github.com/microsoft/azure-container-apps/issues/1219) :-(

**Comment by jason-berk-k1x** [comment]
I have so many issues with ACA and get little to no support....I don't know where one issue ends and the next one starts.

**Comment by fritzfs** [comment]
I agree. I also experienced only pain since the beginning and starting to regret not going for AKS ... :-/
In my case, root cause was somehow hitting the disk space on Consumption profile for ACA ENV :-(

**Comment by jasonrberk** [comment]
That's my issue......AKS is *so* much more complicated.  If ACA just did what it advertised it would be an obvious replacement..... especially on the consumption plan

**Comment by PhilippRoessner** [comment]
## Still experiencing this issue - January 2026

I can confirm this bug is **still present** in production as of January 2026. The `registries.identity` property in Bicep/ARM templates does not propagate correctly for Container Apps Jobs.

### Environment
- **Region**: Germany West Central
- **API Version**: `2024-03-01`
- **Deployment Method**: Bicep via Azure CLI

### Bicep Configuration (correct, but fails)

```bicep
resource containerAppJob 'Microsoft.App/jobs@2024-03-01' = {
  name: 'my-scheduled-job'
  location: location
  identity: {
    type: 'UserAssigned'
    userAssignedIdentities: {
      '${managedIdentityId}': {}
    }
  }
  properties: {
    environmentId: containerAppsEnvironment.id
    configuration: {
      triggerType: 'Schedule'
      scheduleTriggerConfig: {
        cronExpression: '0 9 * * *'
      }
      registries: [
        {
          server: '${acrName}.azurecr.io'
          identity: managedIdentityId  // ← This is correctly set
        }
      ]
    }
    template: {
      containers: [
        {
          name: 'job-container'
          image: '${acrName}.azurecr.io/my-image:latest'
        }
      ]
    }
  }
}
```

### Verification that identity has correct permissions

```bash
# AcrPull role is correctly assigned
az role assignment list --assignee <identity-principal-id> --scope /subscriptions/.../registries/<acr-name>
# Returns: AcrPull role assignment ✅
```

### Symptoms

1. ✅ Bicep deployment succeeds with no errors
2. ✅ Managed identity appears assigned to the job
3. ❌ Job execution fails with `ImagePullBackOff` / `401 Unauthorized`
4. ❌ `az containerapp job show` reveals the registry identity is not properly configured

### Workaround (proves it's a platform bug)

Running the CLI command **immediately after** Bicep deployment fixes the issue:

```bash
az containerapp job update \
    --name my-scheduled-job \
    --resource-group MyResourceGroup \
    --registry-identity /subscriptions/.../userAssignedIdentities/MyIdentity
```

After running this command, the **exact same job** with the **exact same identity** works perfectly. This proves:
- The identity configuration is correct
- The AcrPull role assignment is correct  
- The bug is in how Bicep/ARM propagates the `registries.identity` to the job

### Business Impact

- **Security concern**: We must run a post-deployment script, creating a window where jobs could fail
- **Operational overhead**: Every deployment requires manual intervention
- **CI/CD complexity**: Cannot rely on pure IaC; need scripted workarounds

### Request

1. **What is the timeline** for fixing this in the ARM API?
2. **Is there a different Bicep syntax** that works? (We've tried multiple API versions)
3. Can this be escalated? Issue #816 was opened June 2023 - this is now 2.5 years unresolved.

Thank you for your attention to this issue.

**Comment by fahad-shafique** [comment]
@PhilippRoessner Have you found any fix or a work around for this?
